lib/models/notification.ts

```diff
@@ -49,55 +49,48 @@ export const NotificationBase = t.interface({
 
 // Email Notification
 
-export const NotificationChannelEmail = t.interface({
-  [NotificationChannelEnum.EMAIL]: t.intersection([
-    t.interface({
-      addressSource: NotificationAddressSource,
-      toAddress: EmailAddress
-    }),
-    t.partial({
-      fromAddress: EmailAddress
-    })
-  ])
-});
+export const NotificationChannelEmail = t.intersection([
+  t.interface({
+    addressSource: NotificationAddressSource,
+    toAddress: EmailAddress
+  }),
+  t.partial({
+    fromAddress: EmailAddress
+  })
+]);
 export type NotificationChannelEmail = t.TypeOf<
   typeof NotificationChannelEmail
 >;
 
 export const EmailNotification = t.interface({
   ...NotificationBase.props,
-  channel: NotificationChannelEmail
+  channel: NotificationChannelEmail,
+  type: t.literal(NotificationChannelEnum.EMAIL)
 });
 export type EmailNotification = t.TypeOf<typeof EmailNotification>;
 
 // Webhook Notification
 
 export const NotificationChannelWebhook = t.interface({
-  [NotificationChannelEnum.WEBHOOK]: t.interface({
-    url: HttpsUrl
-  })
+  url: HttpsUrl
 });
 export type NotificationChannelWebhook = t.TypeOf<
   typeof NotificationChannelWebhook
 >;
 
 export const WebhookNotification = t.interface({
   ...NotificationBase.props,
-  channel: NotificationChannelWebhook
+  channel: NotificationChannelWebhook,
```

Comment:
same here, it looks like having the channel specific attributes inside `channel` can be avoided, getting rid of the extra type `NotificationChannelWebhook`